airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -193,6 +193,206 @@ def __init__(self, service, project_id):
+                'compression': compression,
+                'ignoreUnknownValues': ignore_unknown_values
+            },
+            'tableReference': {
+                'projectId': project_id,
+                'datasetId': dataset_id,
+                'tableId': external_table_id,
+            }
+        }
+
+        if schema_fields:
+            table_resource['externalDataConfiguration'].update({
+                'schema': {
+                    'fields': schema_fields
+                }
+            })
+
+        self.log.info('Creating external table: %s', external_project_dataset_table)
+
+        if max_bad_records:
+            table_resource['externalDataConfiguration']['maxBadRecords'] = max_bad_records
+
+        # if following fields are not specified in src_fmt_configs,
+        # honor the top-level params for backward-compatibility
+        if 'skipLeadingRows' not in src_fmt_configs:
+            src_fmt_configs['skipLeadingRows'] = skip_leading_rows
+        if 'fieldDelimiter' not in src_fmt_configs:
+            src_fmt_configs['fieldDelimiter'] = field_delimiter
+        if quote_character:
+            src_fmt_configs['quote'] = quote_character
+        if allow_quoted_newlines:
+            src_fmt_configs['allowQuotedNewlines'] = allow_quoted_newlines
+        if allow_jagged_rows:
+            src_fmt_configs['allowJaggedRows'] = allow_jagged_rows
+
+        src_fmt_to_param_mapping = {
+            'CSV': 'csvOptions',
+            'GOOGLE_SHEETS': 'googleSheetsOptions',
+            'AVRO': 'AVRO',
+            'NEWLINE_DELIMITED_JSON': 'NEWLINE_DELIMITED_JSON',
+            'DATASTORE_BACKUP': 'DATASTORE_BACKUP'
+        }
+
+        src_fmt_to_configs_mapping = {
+            'csvOptions': [
+                'allowJaggedRows', 'allowQuotedNewlines',
+                'fieldDelimiter', 'skipLeadingRows',
+                'quote'
+            ],
+            'NEWLINE_DELIMITED_JSON': [],
+            'AVRO': [],
+            'googleSheetsOptions': ['skipLeadingRows'],
+            'DATASTORE_BACKUP': []
+        }
+
+        valid_configs = src_fmt_to_configs_mapping[
+            src_fmt_to_param_mapping[source_format]
+        ]
+        src_fmt_configs = {
+            k: v
+            for k, v in src_fmt_configs.items() if k in valid_configs
+        }
+
+        table_resource['externalDataConfiguration'][src_fmt_to_param_mapping[
+            source_format]] = src_fmt_configs
+
+        try:
+            self.service.tables().insert(
+                projectId=project_id,
+                datasetId=dataset_id,
+                body=table_resource
+            ).execute()
+
+            self.log.info('External table created successfully: %s',
+                          external_project_dataset_table)
+
+            return True
+
+        except HttpError as err:
+
+            if err.resp.status in [500, 503]:
+                self.log.info('%s: Retryable error, waiting for job to complete.',
```

Comment:
I don't think tables().insert is retry-able, it doesn't return a job id in the response, but instead returns an id to identify the table (which can be returned to the caller). Time.sleep is also not needed since there is no polling/retry.

Reply:
Done. Amended